mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/query/CriterionConverter.java

```diff
@@ -87,26 +88,41 @@ private void registerCriterionConverters() {
     private <T extends Criterion> void registerCriterionConverter(Class<T> type, Function<T, QueryBuilder> f) {
         criterionConverterMap.put(type, (Function<Criterion, QueryBuilder>) f);
     }
-    
+
+    private QueryBuilder createHeaderNameQuery(String headerName) {
```

Comment:
Ok I thought this was not a code issue but a admin issue?
Moreover to have a code that handle old data we will need two indexes.. I am not saying that it is not feasable but it's a topic the team should discuss in order to agree on how to handle that